projects/common/src/utilities/types/types.ts

```diff
@@ -35,3 +35,4 @@ export interface Json {
 export type JsonValue = Json | Json[] | string | number | boolean | null;
 export type PrimitiveValue = string | number | boolean;
 export type ValueOrNull<T> = T | null;
+export type ValueOrUndefined<T> = T | undefined;
```

Comment:
How's this differ from `?: T`?

Reply:
its the same. I feel this is just a bit more cleaner when we have to return something from a function or use this as a type instead of doing ` | undefined`